phoenix-queryserver/src/main/java/org/apache/phoenix/queryserver/register/ZookeeperRegistry.java

```diff
@@ -0,0 +1,73 @@
+/**
+ *
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.phoenix.queryserver.register;
+
+
+import org.apache.commons.logging.Log;
+import org.apache.commons.logging.LogFactory;
+import org.apache.curator.framework.CuratorFramework;
+import org.apache.curator.framework.CuratorFrameworkFactory;
+import org.apache.curator.retry.ExponentialBackoffRetry;
+import org.apache.curator.utils.CloseableUtils;
+import org.apache.phoenix.loadbalancer.service.LoadBalancerConfiguration;
+import org.apache.phoenix.loadbalancer.service.PhoenixQueryServerNode;
+import org.apache.phoenix.queryserver.server.QueryServer;
+import org.apache.zookeeper.CreateMode;
+import org.apache.zookeeper.data.Stat;
+import java.net.InetAddress;
+import java.nio.charset.StandardCharsets;
+
+
+public class ZookeeperRegistry implements Registry {
+
+    protected static final Log LOG = LogFactory.getLog(QueryServer.class);
+    private CuratorFramework client;
+
+    public ZookeeperRegistry(){}
+
+    @Override
+    public  void registerServer(LoadBalancerConfiguration loadBalancerConfiguration, int avaticaServerPort,
+                                String zookeeperConnectString)
+            throws Exception {
+
+        this.client = CuratorFrameworkFactory.newClient(zookeeperConnectString,
+                new ExponentialBackoffRetry(1000,10));
+        this.client.start();
+        String host = InetAddress.getLocalHost().getHostName();
```

Comment:
I think it makes sense to have the caller pass this info in. Or, provide some information object on the `QueryServer` instance.